build.gradle

```diff
@@ -12,7 +12,7 @@ import de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin
 import org.apache.tools.ant.taskdefs.condition.Os
 import org.elasticsearch.gradle.internal.BuildPlugin
 import org.elasticsearch.gradle.Version
-import org.elasticsearch.gradle.internal.VersionProperties
+import org.elasticsearch.gradle.VersionProperties
```

Comment:
Public plugin still requires VersionProperties. We should revisit general version handling and support in external plugins later again

Reply:
Indeed. Really the only thing we use it for is to determine which version of the elasticsearch dependencies to use. We could do a few things here:

1. Just extract the version number (perhaps at build time) instead of bringing in VersionProperties and all it's friends.
2. Don't make any assumptions about ES version at all and let the plugin author specify it. That _could_ get weird if there are requirements but generally there's no reason you can't use 8.0 build-tools to build a 7.x plugin.

Reply:
I think option 2) is the way to go and we could publish a BOM as part of our releases and consume these elasticsearch related dependencies via as a platform. IMO this is the most natural way to deal with those kind of dependencies in the plugin. Again, I'd like to tackle this in a separate PR once we actually split out the build-tool project.